tests/parametric/test_otel_span_methods.py

```diff
@@ -71,7 +71,7 @@ def test_otel_set_service_name(self, test_agent, test_library):
         context.library == "java",
         reason="Old array encoding was removed in 1.22.0 and new span naming introduced in 1.24.0: no version elligible for this test.",
     )
-    @irrelevant(context.library < "golang@v1.59.0", reason="Old array encoding no longer supported")
+    @irrelevant(context.library >= "golang@v1.59.0", reason="New span naming introduced in v1.59.0")
```

Comment:
Does it mean that `test_otel_set_attributes_different_types` test an old behavior that is not used by the go tracer, starting version `1.59.0` included ? if yes, all good !

It may be a good idea to rename the test method `test_otel_set_attributes_different_types_legacy`, with a small comment explaining where to find the test for the new behavior.

Reply:
> Does it mean that test_otel_set_attributes_different_types test an old behavior that is not used by the go tracer, starting version 1.59.0 included ?

Yes that's correct.

Reply:
> It may be a good idea to rename the test method test_otel_set_attributes_different_types_legacy, with a small comment explaining where to find the test for the new behavior.

Done